plugins/modules/vault_kv2_write.py

```diff
@@ -0,0 +1,221 @@
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+module: vault_kv2_write
+version_added: 4.2.0
+author:
+  - Devon Mar (@devon-mar)
+short_description: Perform a write operation against a KVv2 secret in HashiCorp Vault
+description:
+  - Perform a write/patch operation against a KVv2 secret in HashiCorp Vault.
+requirements:
+  - C(hvac) (L(Python library,https://hvac.readthedocs.io/en/stable/overview.html))
+  - For detailed requirements, see R(the collection requirements page,ansible_collections.community.hashi_vault.docsite.user_guide.requirements).
+extends_documentation_fragment:
+  - community.hashi_vault.attributes
+  - community.hashi_vault.attributes.action_group
+  - community.hashi_vault.connection
+  - community.hashi_vault.auth
+  - community.hashi_vault.engine_mount
+attributes:
+  check_mode:
+    support: partial
+    details:
+      - If I(read) is C(true), full check mode functionality is supported.
+      - If I(read) is C(false), the status will always be C(changed) but a write will not be performed in check mode.
+options:
+  engine_mount_point:
+    type: str
+    default: secret
+  path:
+    type: str
+    required: true
+    description:
+      - Vault KVv2 path to be written to.
+      - This is relative to the I(engine_mount_point), so the mount path should not be included.
+  data:
+    type: dict
+    required: true
+    description:
+      - KVv2 secret data to write/patch.
+  cas:
+    type: int
+    description:
+      - Perform a check-and-set operation.
+  read:
+    type: bool
+    default: false
+    description:
+      - Read the secret first and write only when I(data) differs from the read data.
+      - Requires C(read) permission on the secret if C(true).
+      - If C(false), this module will always write to I(path).
+"""
+
+EXAMPLES = r"""
+- name: Write/create a secret
+  community.hashi_vault.vault_kv2_write:
+    url: https://vault:8201
+    path: hello
+    data:
+      foo: bar
+
+- name: Write with check and set
+  community.hashi_vault.vault_kv2_write:
+    url: https://vault:8201
+    path: caspath
+    cas: 0
+    data:
+      foo: bar
+"""
+
+RETURN = r"""
+raw:
+  type: dict
+  description: The raw Vault response.
+  returned: changed
+  sample:
+    auth:
+    data:
+      created_time: "2023-02-21T19:51:50.801757862Z"
+      custom_metadata:
+      deletion_time: ""
+      destroyed: false
+      version: 1
+    lease_duration: 0
+    lease_id: ""
+    renewable: false
+    request_id: 52eb1aa7-5a38-9a02-9246-efc5bf9581ec
+    warnings: null
+    wrap_info: null
+"""
+
+import traceback
+
+from ansible.module_utils._text import to_native
+from ansible.module_utils.basic import missing_required_lib
+from ansible_collections.community.hashi_vault.plugins.module_utils._hashi_vault_common import (
+    HashiVaultValueError,
+)
+from ansible_collections.community.hashi_vault.plugins.module_utils._hashi_vault_module import (
```

Comment:
this is very minor, but I'm starting to prefer relative imports, and I'm changing existing absolute imports to relative when changes are made to existing files. Let's update these to that form (I don't know offhand the right relative level sorry for not including the suggestion).